roles/translator/src/lib/sv2/channel_manager/message_handler.rs

```diff
@@ -0,0 +1,426 @@
+                    let new_downstream_extended_channel = ExtendedChannel::new(
+                        m.channel_id,
+                        user_identity.clone(),
+                        new_extranonce_prefix.clone().into_static().to_vec(),
+                        target.clone().into(),
+                        nominal_hashrate,
+                        true,
+                        new_extranonce_size,
+                    );
+                    channel_manager_data.extended_channels.insert(
+                        m.channel_id,
+                        Arc::new(RwLock::new(new_downstream_extended_channel)),
+                    );
+                    let new_open_extended_mining_channel_success = OpenExtendedMiningChannelSuccess {
+                        request_id: m.request_id,
+                        channel_id: m.channel_id,
+                        extranonce_prefix: new_extranonce_prefix,
+                        extranonce_size: new_extranonce_size,
+                        target: m.target.clone(),
+                    };
+                    return new_open_extended_mining_channel_success.into_static();
+                }
+            }
+
+            // If we are not in aggregated mode, we just insert the extended channel into the map
+            channel_manager_data.extended_channels
+                .insert(m.channel_id, Arc::new(RwLock::new(extended_channel)));
+
+            m.into_static()
+        }).unwrap();
+
+        self.channel_state
+            .sv1_server_sender
+            .send(Mining::OpenExtendedMiningChannelSuccess(success.clone()))
+            .await
+            .map_err(|e| {
+                error!("Failed to send OpenExtendedMiningChannelSuccess: {:?}", e);
+                HandlerError::External(Box::new(TproxyError::ChannelErrorSender))
+            })?;
+
+        Ok(())
+    }
+
+    async fn handle_open_mining_channel_error(
+        &mut self,
+        m: roles_logic_sv2::mining_sv2::OpenMiningChannelError<'_>,
+    ) -> Result<(), HandlerError> {
+        error!(
+            "Received OpenExtendedMiningChannelError from upstream with error code {}",
+            std::str::from_utf8(m.error_code.as_ref()).unwrap_or("unknown error code")
+        );
+        todo!("OpenExtendedMiningChannelError not handled yet");
+    }
+
+    async fn handle_update_channel_error(
+        &mut self,
+        m: roles_logic_sv2::mining_sv2::UpdateChannelError<'_>,
+    ) -> Result<(), HandlerError> {
+        error!(
+            "Received UpdateChannelError from upstream with error code {}",
+            std::str::from_utf8(m.error_code.as_ref()).unwrap_or("unknown error code")
+        );
+        Ok(())
+    }
+
+    async fn handle_close_channel(
+        &mut self,
+        m: roles_logic_sv2::mining_sv2::CloseChannel<'_>,
+    ) -> Result<(), HandlerError> {
+        info!(
+            "Received CloseChannel from upstream for channel id: {}",
+            m.channel_id
```

Comment:
here, we would only be showing`channel_id`, but not other relevant message fields

we have `Display` trait implementations for all messages now, we should leverage them
```suggestion
            "Received: {}", m
```